File: 77/source.php
<?php

	$primes  = [];
	$primes2 = [];
	$allNum  = 1000000;
	$ways    = [];
	$ways[0] = 1;

	function fillArray($len){
	global $ways;
		for($i = 0; $i < $len; $i++){
			array_push($ways, 0);
		}

	}

	/* filling $num primes to array */
	function fillingPrimes($num){
	global $primes;
		for($i=0;$i<$num;$i++)
			{
			$primes[$i] = (!($i == 2 || $i%2 == 0 || $i == 0 || $i == 1));
			}
		for($i=3;$i<=($num/2);$i=$i+2)
			{
			if($primes[$i])
				{
				for($j=2*$i;$j<$num;$j+=$i)
					{
					$primes[$j]=false;
					}
				}
			}
	$primes[2] = true;
	}

	function copyPrimes($num){
	global $primes, $primes2;
		for($i=0; $i < $num; $i++){
			if($primes[$i] == 1){
				array_push($primes2, $i);
			}
		}
	}

	function calculate($target){
	global $ways, $primes2;
		$len      = count($primes2);
		$retValue = 0;
		for($i = 0; $i < $len; $i++){
			for($j = $primes2[$i]; $j <= $target; $j++){
				$tmp = $j - $primes2[$i];
				$ways[$j] += $ways[$tmp];
				$retValue  = $ways[$j];
			}
		}
		return $retValue;
	}


	function main(){
	global $primes2, $allNum, $ways;
		$result = 0;
		echo"\nFilling Primes\n";
		fillingPrimes($allNum);
		copyPrimes($allNum);
		echo"\n Looking for the first value which can be written as the sum of primes in over five thousand different ways \n";
		fillArray(6000);
		for($i = 1; $i < 5000; $i++){
			$wys = calculate($i);
			//echo"\n Possible to sum {$i} on {$wys} ways";
			if($wys > 4999){
				$result = $i;
				break;
			}
			/* cleaning array $ways for new calculating */
			for($j = 1; $j < count($ways); $j++){
				$ways[$j] = 0;
			}
		}
		echo"\nResult: {$result}\n";

	}


	main();


?>

File: 43/source.py
from itertools import permutations

def substringDivisibility(num):
    num=str(num)
    d1 = int(num[1:4])
    d2 = int(num[2:5])
    d3 = int(num[3:6])
    d4 = int(num[4:7])
    d5 = int(num[5:8])
    d6 = int(num[6:9])
    d7 = int(num[7:10])
    if d1%2 != 0:
        return False
    if d2%3 != 0:
        return False
    if d3%5 != 0:
        return False
    if d4%7 != 0:
        return False
    if d5%11 != 0:
        return False
    if d6%13 != 0:
        return False
    if d7%17 != 0:
        return False
    return True

result = set()
for i in permutations([1,2,3,4,5,6,7,8,9,0]):
    n = int(''.join(map(str,i)))
    if substringDivisibility(n):
        result.add(n)
print("Result: ", sum(result))